packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -155,3 +159,23 @@ class _MergingListenable extends ChangeNotifier {
     super.dispose();
   }
 }
+
+/// A [ChangeNotifier] that holds a single value.
+///
+/// When [value] is replaced, this class notifies its listeners.
+class ValueNotifier<T> extends ChangeNotifier {
+  /// Creates a [ChangeNotifier] that wraps this value.
+  ValueNotifier(this._value);
+
+  /// The current value stored in this notifier.
+  ///
+  /// When the value is replaced, this class notifies its listeners.
+  T get value => _value;
+  T _value;
+  set value(T newValue) {
+    if (_value == newValue)
```

Comment:
This will sadly go pathological for `ValueNotifier<double>` when you give it a `NaN`, but I guess there's no good solution to that.